### DISCUSSION TITLE
EVSE support

### DISCUSSION THREAD

**Discussion by Steve-Mcl** [matter-js/matter.js#1440] [Q&A]
Hi, first some big thanks for a great project. I love seeing the progress being made.

I have had a quick search and scan of the code but can see no mention of EVSE Support.

I am not super knowledgeable on the specification but I am lead to believe EVSE is defined in the 1.3 spec?

Is this something matter js will add support for?

Thanks, Steve.

**Comment by Apollon77** [comment]
All clusters and device types for EVSE are available in matter.js as of Matter 1.3, so even if there are no "default implementations" for potentially relevant clusters you can build EVSE devices with it or use them as controller.
But please also note that e.g. "Device Energy Management" cluster (and others or certain features out of the relevant ones) were provisional in Matter 1.3 and were just finalized in Matter 1.4. But in this regard anything which was supported and defined in Matter 1.3 is available.

What exactly you miss?

**Comment by Steve-Mcl** [comment]
> What exactly you miss?

So I was looking to raise a PR against Sams work in https://github.com/sammachin/node-red-matter-bridge for adding EVSE Node to the Node-RED palette. 

For example, the package builds an `OnOffLightDevice` Device here: https://github.com/Steve-Mcl/node-red-matter-bridge/blob/92369d1f5c68e8b895e36623814d367d001833f3/devices/onofflight.js#L3C88-L3C104 so I went looking for an EVSE _thing_ in the matter js package and when I didnt see it I have assumed it was not supported.

As you can likely tell from my language and description of all this, i am not up-to-speed on how Matter devices are put together or what components are required. And as you say, all of the elements are there to make/support EVSE, so it is my most likely my knowledge in this area that is thing that is missing :)

Any direction or cues based on the above link and my desire to add EVSE support in the same style would be a great help.

**Comment by sammachin** [comment]
EVSE is on my list to add (I've got an OCPP enabled EVSE at home that I'd like to control)
But at the moment there's no support for EVSE in any of the major ecosystems* so it seems a bit pointless to implement.

* Home Assistant has it I believe but there are easier/better ways to connect Home Assistant & Node-RED than with Matter hence I don't consider it.

**Comment by Apollon77** [comment]
In general one dev tried and found an issue (https://github.com/project-chip/matter.js/issues/1455) ... so we will be there soon :-)

**Reply by Apollon77** [reply]
@Steve-Mcl please try the latest nightly build (npm tag dev). The mode should now work as expected.
